codegen/src/types/type_def_params.rs

```diff
@@ -0,0 +1,76 @@
+// Copyright 2019-2022 Parity Technologies (UK) Ltd.
+// This file is part of subxt.
+//
+// subxt is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// subxt is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with subxt.  If not, see <http://www.gnu.org/licenses/>.
+
+use super::TypeParameter;
+use crate::types::CompositeDefField;
+use quote::quote;
+use std::collections::HashSet;
+
+#[derive(Clone, Debug, Default)]
+pub struct TypeDefParameters {
```

Comment:
SOme docs would be good here. Maybe on on the module level as well.

Reply:
Done